`tests/test_tibs.py::test_from_large_ints` failing: `pyo3_runtime.PanicException: attempt to negate with overflow`

On top of 0.5.6 / 6e569ee32403639467a1cde3b83a01af96bbfd9c:

```pytb
$ uv venv -p 3.13
Using CPython 3.13.12 interpreter at: /usr/bin/python3.13
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
$ uv pip install --config-settings=build-args=--profile=dev -e . pytest hypothesis
Resolved 8 packages in 319ms
      Built tibs @ file:///tmp/tibs
Prepared 1 package in 16.54s
Installed 8 packages in 16ms
 + hypothesis==6.151.9
 + iniconfig==2.3.0
 + packaging==26.0
 + pluggy==1.6.0
 + pygments==2.19.2
 + pytest==9.0.2
 + sortedcontainers==2.4.0
 + tibs==0.5.6 (from file:///tmp/tibs)
$ RUST_BACKTRACE=1 .venv/bin/pytest tests/test_tibs.py 
========================================================= test session starts =========================================================
platform linux -- Python 3.13.12, pytest-9.0.2, pluggy-1.6.0
rootdir: /tmp/tibs
configfile: pyproject.toml
plugins: hypothesis-6.151.9
collected 22 items                                                                                                                    

tests/test_tibs.py ............F.........                                                                                       [100%]

============================================================== FAILURES ===============================================================
________________________________________________________ test_from_large_ints _________________________________________________________

    def test_from_large_ints():
        with pytest.raises(ValueError):
            _ = Tibs.from_i(-1, 1000)
>       a = Tibs.from_i(-1, 128)
            ^^^^^^^^^^^^^^^^^^^^
E       pyo3_runtime.PanicException: attempt to negate with overflow

tests/test_tibs.py:97: PanicException
-------------------------------------------------------- Captured stderr call ---------------------------------------------------------

thread '<unnamed>' (336597) panicked at src/helpers.rs:506:19:
attempt to negate with overflow
stack backtrace:
   0: __rustc::rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic_const::panic_const_neg_overflow
   3: tibs::helpers::bv_from_i128
             at ./src/helpers.rs:506:19
   4: tibs::tibs_::Tibs::from_i
             at ./src/tibs_.rs:481:18
   5: tibs::tibs_::Tibs::__pymethod_from_i__
             at ./src/tibs_.rs:188:1
   6: pyo3::impl_::trampoline::cfunction_with_keywords::inner::{{closure}}
             at /home/mgorny/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pyo3-0.28.2/src/impl_/trampoline.rs:82:42
   7: pyo3::impl_::trampoline::trampoline::{{closure}}
             at /home/mgorny/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pyo3-0.28.2/src/impl_/trampoline.rs:269:54
   8: std::panicking::catch_unwind::do_call
             at /usr/lib/rust/1.94.0/lib/rustlib/src/rust/library/std/src/panicking.rs:581:40
   9: __rust_try
  10: std::panicking::catch_unwind
             at /usr/lib/rust/1.94.0/lib/rustlib/src/rust/library/std/src/panicking.rs:544:19
  11: std::panic::catch_unwind
             at /usr/lib/rust/1.94.0/lib/rustlib/src/rust/library/std/src/panic.rs:359:14
  12: pyo3::impl_::trampoline::trampoline
             at /home/mgorny/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pyo3-0.28.2/src/impl_/trampoline.rs:269:9
  13: pyo3::impl_::trampoline::cfunction_with_keywords::inner
             at /home/mgorny/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pyo3-0.28.2/src/impl_/trampoline.rs:82:26
  14: pyo3::impl_::trampoline::cfunction_with_keywords
             at /home/mgorny/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pyo3-0.28.2/src/impl_/trampoline.rs:72:22
  15: <unknown>
  16: _PyObject_MakeTpCall
  17: _PyEval_EvalFrameDefault
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: _PyObject_MakeTpCall
  22: _PyEval_EvalFrameDefault
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: _PyEval_EvalFrameDefault
  28: <unknown>
  29: <unknown>
  30: <unknown>
  31: _PyObject_MakeTpCall
  32: _PyEval_EvalFrameDefault
  33: <unknown>
  34: <unknown>
  35: <unknown>
  36: _PyObject_MakeTpCall
  37: _PyEval_EvalFrameDefault
  38: <unknown>
  39: <unknown>
  40: <unknown>
  41: _PyObject_MakeTpCall
  42: _PyEval_EvalFrameDefault
  43: PyEval_EvalCode
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: Py_RunMain
  50: Py_BytesMain
  51: <unknown>
  52: __libc_start_main
  53: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
======================================================= short test summary info =======================================================
FAILED tests/test_tibs.py::test_from_large_ints - pyo3_runtime.PanicException: attempt to negate with overflow
==================================================== 1 failed, 21 passed in 0.19s =====================================================
```

--- a/src/helpers.rs
+++ b/src/helpers.rs
@@ -498,7 +498,7 @@ pub(crate) fn bv_from_u128(value: u128, length: i64) -> PyResult<BV> {
 
 #[inline]
 pub(crate) fn bv_from_i128(value: i128, length: i64) -> PyResult<BV> {
-    if length <= 0 || length > 128 {
+    if length <= 0 || length >= 128 {
         return Err(PyValueError::new_err(format!(
             "Bit length for signed int must be between 1 and 128. Received {length}."
         )));

--- a/tests/test_tibs.py
+++ b/tests/test_tibs.py
@@ -94,7 +94,7 @@ def test_from_i_errors():
 def test_from_large_ints():
     with pytest.raises(ValueError):
         _ = Tibs.from_i(-1, 1000)
-    a = Tibs.from_i(-1, 128)
+    a = Tibs.from_i(-1, 127)
     assert a.all()
     with pytest.raises(ValueError):
         _ = Mutibs.from_i(-1, 1000)

